clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -674,10 +674,14 @@ std::string PointerVariableConstraint::gatherQualStrings(void) const {
 std::string PointerVariableConstraint::mkString(Constraints &CS,
                                                 bool EmitName, bool ForItype,
                                                 bool EmitPointee,
-                                                bool UnmaskTypedef) const {
+                                                bool UnmaskTypedef,
+                                                std::string UseName) const {
+  if (UseName.empty())
+    UseName = getName();
+
   if (IsTypedef && !UnmaskTypedef) {
     return gatherQualStrings() + TypedefString +
-           (EmitName && getName() != RETVAR ? (" " + getName()) : " ");
+           (EmitName && UseName != RETVAR ? (" " + UseName) : " ");
```

Comment:
Is there a risk here that the trick we are playing with `RETVAR` will get messed up if `UseName` comes in clashing with that name? Maybe that was also possible before.

Reply:
Proposed change: at the top of this function
```c
// The name field encodes if this variable is the return type for a function.   
// TODO: store this information in a separate field.
bool IsReturn = getName() == RETVAR;
```
and then all subsequent comparisons to `RETVAR` look at `IsReturn` instead. This way it still only depends on the `Name` field of this instance of `PointerVariableConstraint`, and we don't have inconsistent usage of `UseName` and `getName()` in the body of the function.

Reply:
I'll commit this